microsoft-edge/webview2/how-to/winrt-from-js.md

```diff
@@ -345,6 +355,96 @@ Next, use the DevTools Console to demonstrate that web-side code can call the in
+1. Have the WinRTAdapter project 'Add a reference' to your new third project containing your WinRT class.
+
+1. Update the WinRTAdapter project's Include filter in the properties to also include your new class.
+
+1. Add an additional line to `InitializeWebView2Async` to add your winrt class's namespace:
+
+   `WebView2.CoreWebView2.AddHostObjectToScript("MyCustomNamespace", dispatchAdapter.WrapNamedObject("MyCustomNamespace", dispatchAdapter));`
+
+1. For easy method calling from the web, optionally add your namespace sync proxy as a global object in script.  For example:
+
+   `window.MyCustomNamespace = chrome.webview.hostObjects.sync.MyCustomNamespace;`
+
+For an example of this, see the following WebView2 sample:
+
+* [uwp-wv2winrt-custom-csharp-winrt](https://github.com/MicrosoftEdge/WebView2Samples/compare/uwp-wv2winrt-custom-csharp-winrt) - A sample, as a branch.
+
+
+<!-- =============================================== -->
+## Asynchronous WinRT methods
+
+Following the steps in the above guide, you should be able to use synchronous proxies. For async method calls, you will need to use `chrome.webview.hostObjects.options.forceAsyncMethodMatches`.
+
+The [forceAsyncMethodMatches](/microsoft-edge/webview2/reference/winrt/microsoft_web_webview2_core/corewebview2#addhostobjecttoscript) property is an array of regexes, where if any regex matches a method name on a sync proxy, the method will be run asynchronously instead. Setting this to `[/Async$/]` will have it match any method ending with the suffix `Async`.  Then matching method calls work just like a method on an async proxy and returns a promise that you can await.
+
+From the above Reference documentation:
+
+| Options property | Details |
+|---|---|
+| `forceAsyncMethodMatches` | This is an array of regular expressions. When calling a method on a synchronous proxy, the method call will be performed asynchronously if the method name matches a string or regular expression in this array. Setting this value to `Async` will make any method that ends with Async be an asynchronous method call. If an async method doesn't match here and isn't forced to be asynchronous, the method will be invoked synchronously, blocking execution of the calling JavaScript and then returning the resolution of the promise, rather than returning a promise. |
+
+Example:
+
+```javascript
+const Windows = chrome.webview.hostObjects.sync.Windows;
+chrome.webview.hostObjects.options.forceAsyncMethodMatches = [/Async$/];
+
+let result = await Windows.System.Launcher.launchUriAsync(new Windows.Foundation.Uri('https://contoso.com/'));
+```
+
+
+<!-- ====================================================================== -->
+
+## Subscribing to WinRT events
+
+WinRT events are also exposed via the script proxies. You can add and remove event handlers for instance and static WinRT events using the `addEventListener(string eventName, function handler)` and `removeEventListener(string eventName, function handler)` methods. 
```

Comment:
Re-reading this the combo of for and instance was confusing me.
```suggestion
WinRT events are also exposed via the script proxies. You can add and remove event handlers of instance and static WinRT events using the `addEventListener(string eventName, function handler)` and `removeEventListener(string eventName, function handler)` methods. 
```

Reply:
You can add and remove event handlers of instance WinRT events and static WinRT events by using the

Reply:
I'm adding my above wording now...